### DISCUSSION TITLE
A place for questions and ideas!

### DISCUSSION THREAD

**Discussion by kinnala** [kinnala/scikit-fem#700] [Announcements]
Github Discussions is a new feature for separating issues and more open ended discussion or Q&A format questions. I figured that this is exactly what we need in scikit-fem.

Hence, I migrated several discussions from Issues to Discussions. There are a few categories to get started with:
- Ideas: If you have any ideas floating around (e.g., vague ideas on new features)
- Q&A: If you have a specific question (how to do this using scikit-fem?)
- Show and tell: If you want to demonstrate cool things you did using scikit-fem
- Others: If none of the above applies

I'll try to moderate the discussion/categories as we go. I modified the README.md to suggest using Discussions instead of Issues for specific questions.

**Comment by rndhara** [comment]
When I'm dealing with one dimensional mesh as follows.
import numpy as np

# one dimensional
m = MeshLine(np.linspace(0, 1, 2)).refined()

# visualize works fine!
from skfem.visuals.matplotlib import draw
draw(m)

e = ElementTriP1()
# precompute global basis at quadrature points
basis = Basis(m, e)  # shorthand for CellBasis

I'm getting the following error. No idea how to fix the error. Please help me out.
![Screenshot 2023-06-27 132408](https://github.com/kinnala/scikit-fem/assets/137891820/007a8ff3-be7a-400b-b51f-00b154a34ef9)

**Reply by gdmcbain** [reply]
```
e = ElementTriP1()
```
ia a triangular element, which is two-dimensional. Try a one-dimensional element.

**Reply by gdmcbain** [reply]
```
e = ElementLineP1()
```
as in [ex15](https://github.com/kinnala/scikit-fem/blob/123deb8ef24c447eb815d8fe193535f62079b714/docs/examples/ex15.py#L9)

**Reply by gdmcbain** [reply]
P. S :—I think this question though would be better asked as a new discussion rather than as a comment on this issue which is about having discussions. Try over here: https://github.com/kinnala/scikit-fem/discussions, as described in the README for [getting help](https://github.com/kinnala/scikit-fem#getting-help).

**Reply by rndhara** [reply]
Thanks for the quick response. Actually, I'm a beginner and sorry to post it here. Furthermore, I couldn't find how and where to look for different elements definitions implemented under scikit-fem.

**Reply by kinnala** [reply]
If you have something like ipython with autocomplete, you can do glob import and start writing "Element" and press TAB.

Otherwise you can check out the docs: https://scikit-fem.readthedocs.io/en/latest/api.html#module-skfem.element

**Reply by rndhara** [reply]
Thanks a lot.
